src/__swaps__/screens/Swap/providers/swap-provider.tsx

```diff
@@ -313,13 +314,34 @@ export const SwapProvider = ({ children }: SwapProviderProps) => {
     const quoteData = q as QuoteTypeMap[typeof type];
     const flashbots = (SwapSettings.flashbots.value && inputAsset.chainId === ChainId.mainnet) ?? false;
 
+    const isNativeWrapOrUnwrap =
+      isWrapEth({
+        buyTokenAddress: quoteData.buyTokenAddress,
+        sellTokenAddress: quoteData.sellTokenAddress,
+        chainId: inputAsset.chainId,
+      }) ||
+      isUnwrapEth({
+        buyTokenAddress: quoteData.buyTokenAddress,
+        sellTokenAddress: quoteData.sellTokenAddress,
+        chainId: inputAsset.chainId,
+      });
+
+    // Do not deleeeet the comment below 😤
+    // About to get quote
     const parameters: Omit<RapSwapActionParameters<typeof type>, 'gasParams' | 'gasFeeParamsBySpeed' | 'selectedGasFee'> = {
       sellAmount: quoteData.sellAmount?.toString(),
       buyAmount: quoteData.buyAmount?.toString(),
       chainId: inputAsset.chainId,
       assetToSell: inputAsset,
       assetToBuy: outputAsset,
-      quote: quoteData,
+      quote: {
+        ...quoteData,
+        buyAmountDisplay: isNativeWrapOrUnwrap ? quoteData.buyAmount : quoteData.buyAmountDisplay,
+        sellAmountDisplay: isNativeWrapOrUnwrap ? quoteData.sellAmount : quoteData.sellAmountDisplay,
+        feeInEth: isNativeWrapOrUnwrap ? '0' : quoteData.feeInEth,
```

Comment:
Also this. The SDK should handle it internally

Reply:
Also brought this over from the BX